internal/server/data/provideruser.go

```diff
@@ -78,22 +90,68 @@ func CreateProviderUser(db GormTxn, provider *models.Provider, ident *models.Ide
 	return pu, nil
 }
 
-func UpdateProviderUser(tx WriteTxn, providerUser *models.ProviderUser) error {
+var ErrSourceOfTruthConflict = fmt.Errorf("cannot update user's email, this user exists in multiple identity providers")
+
+func UpdateProviderUser(tx GormTxn, providerUser *models.ProviderUser) error {
 	if err := validateProviderUser(providerUser); err != nil {
 		return err
 	}
 	providerUser.LastUpdate = time.Now().UTC()
 
+	// check if the user's email has changed
+	identity, err := GetIdentity(tx, ByID(providerUser.IdentityID), Preload("Providers"))
+	if err != nil {
+		return fmt.Errorf("check identity before provider update: %w", err)
+	}
+
+	if identity.Name != providerUser.Email {
```

Comment:
It seems a workaround for the email conflict is to save the alternate email to only the `ProviderUser` record and report it as an alias

Reply:
That would be a potentially large change so I think we should address it in a follow-up.